silkworm/db/datastore/snapshots/snapshot_repository.cpp

```diff
@@ -246,7 +267,7 @@ SnapshotPathList SnapshotRepository::get_files(std::string_view ext) const {
     return snapshot_files;
 }
 
-std::vector<StepRange> SnapshotRepository::list_dir_file_ranges() const {
+std::set<StepRange> SnapshotRepository::list_dir_file_ranges() const {
```

Comment:
I suggest to refactor it to something like:
```suggestion
void SnapshotRepository::list_files(std::function<void(SnapshotPath)> consumer) const {
```
It makes the function more reusable: we can use it whether we want if we want duplicates/sorting or not

This is optional though, and can be refactored later.

Reply:
I'll leave it for later